internal/job/OrdersStatusJob.go

```diff
@@ -0,0 +1,68 @@
+package job
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/andrey67895/go_diplom_first/internal/config"
+	"github.com/andrey67895/go_diplom_first/internal/database"
+	"github.com/andrey67895/go_diplom_first/internal/helpers"
+	"github.com/andrey67895/go_diplom_first/internal/model"
+)
+
+var client http.Client
+
+func GetAndUpdateOrderStatusByAccrual(login string, number string) (*http.Response, error) {
+	url := config.AccrualSystemAddress + "/api/orders/" + number
+	r, _ := http.NewRequest(http.MethodGet, url, nil)
+	body, err := client.Do(r)
+	if err != nil {
+		helpers.TLog.Error(err.Error())
+	}
+	defer body.Body.Close()
+	if body.StatusCode == http.StatusOK {
+		var tModel model.OrdersAccrualModel
+		err = json.NewDecoder(body.Body).Decode(&tModel)
+		if err != nil {
+			helpers.TLog.Error(err.Error())
+			return body, err
+		}
+		if *tModel.Status == "PROCESSING" || *tModel.Status == "INVALID" || *tModel.Status == "PROCESSED" {
+			err := database.DBStorage.UpdateOrders(tModel, *tModel.Status == "PROCESSED", login)
+			if err != nil {
+				helpers.TLog.Error(err.Error())
+				return body, err
+			}
+		}
+	}
+	return body, nil
+}
+
+func OrdersStatusJob() {
```

Comment:
```suggestion
func OrdersStatusJob(ctx context.Context, wg sync.WaitGroup) {
```
Для функции стоит предусмотреть graceful shutdown. Например, передать внутрь контекст, сгенеренный на старте приложения и `wg sync.WaitGroup`. Вместо `for range` использовать `for select`. А по выходу (или через defer) дергать wg.Done().

Reply:
готово